flytekit/clis/sdk_in_container/run.py

```diff
@@ -48,10 +60,51 @@ def convert(self, value, param, ctx) -> typing.Optional[typing.Union[typing.Dict
 class DataframeType(click.ParamType):
     name = "dataframe"
 
-    def convert(self, value, param, ctx) -> pd.DataFrame:
-        if not ctx.obj[REMOTE_KEY]:
+    def __init__(self, input_type: typing.Type[StructuredDataset]):
+        self._sdt = TypeEngine.to_literal_type(input_type)
+
+    def convert(self, value, param, ctx) -> typing.Union[pd.DataFrame, str]:
+        if not ctx.obj[REMOTE_FLAG_KEY]:
             return pd.read_parquet(value)
 
+        # The value here should be a string containing a path to a parquet file. If not running locally, then we have
+        # no need of reading the parquet file.
+        # This relies on the TypeEngine to trigger the remote encoder.
+        return StructuredDataset(
```

Comment:
should it be a structured dataset, dont we have to upload it yet? and that has to be done using the signed url right?

Reply:
in this instance the type engine will take care of uploading it yes.